mutiny-core/src/node.rs

```diff
@@ -449,24 +427,24 @@ impl<S: MutinyStorage> Node<S> {
             }
         });
 
-        // set up the connection handlers to start in the background
-        let self_connection = PubkeyConnectionInfo {
+        #[cfg(target_arch = "wasm32")]
+        start_reconnection_handling(
+            &persister.storage,
             pubkey,
-            connection_type: ConnectionType::Mutiny(websocket_proxy_addr.to_string()),
-            original_connection_string: format!("mutiny:{pubkey}@{websocket_proxy_addr}"),
-        };
-        let multi_socket = Arc::new(Mutex::new(MultiWsSocketDescriptor::new(
+            websocket_proxy_addr.clone(),
             peer_man.clone(),
-            pubkey.serialize().to_vec(),
+            &logger,
+            uuid.clone(),
+            &lsp_client,
             stop.clone(),
-            logger.clone(),
-        )));
+            stopped_components.clone(),
+        )
+        .await?;
+
+        #[cfg(not(target_arch = "wasm32"))]
```

Comment:
here too, the only diff is the `websocket_proxy_addr` I think you can just put that behind a build config